src/Symfony/Component/PropertyInfo/Tests/Fixtures/TraitUsage/DummyTrait.php

```diff
@@ -11,46 +11,9 @@
 
 namespace Symfony\Component\PropertyInfo\Tests\Fixtures\TraitUsage;
 
-use Symfony\Component\PropertyInfo\Tests\Fixtures\Dummy;
+use Symfony\Component\PropertyInfo\Tests\Fixtures\DummyTraitExternal;
 
 trait DummyTrait
 {
-    /**
```

Comment:
Instead of moving everywhere to another trait, maybe we can keep a few here to cover more cases?

Reply:
I am not sure to follow what you would like.

I had to move the trait to an another namespace for proof testing.

Also this structure cover more than before without losing anything to me, as the code and test logic are the same, only the recursive fact was missing.

I may create an another trait for the recursive case to let the actual untouched, but I think I will have to double the test suite with not much more reliability value.

Reply:
I think what Fabien means is that some methods should remain in this trait, while others are moved to the recursive one.

Reply:
I didn't get the point about partial moving, however, I put an another commit making the global test diff clearer.

Tell me if this version is OK to you. :+1: